### DISCUSSION TITLE
Multiply scoop locations / Portable scoop

### DISCUSSION THREAD

**Discussion by hagaigold** [ScoopInstaller/Scoop#5675] [Ideas]
Hello,

Maybe this is out of the scope of the amazing scoop project, but I wonder if someone tried to use scoop for a limited scope. 
Think of modules dependencies for a project that I want to be "locally" to the project instead to install it in the "all" user space.

A real-life example: Lua project scripts for [Clink](https://chrisant996.github.io/clink/).
I aware about git submodules and alike- I just like the simplicity of scoop to fetch anything from git.

happy day

**Comment by rashil2000** [comment]
This will be achievable once https://github.com/ScoopInstaller/Scoop/issues/1813 gets fixed

**Comment by hagaigold** [comment]
> This will be achievable once #1813 gets fixed

if I understand #1813- it is about create a %SCOOP_PATH% env variable instead of manipulating the user %PATH% again & again.  

My Intention is to "localize" the apps to a project scope:
`%USERPROFILE%\scoop\apps`, `projectA\scoop\apps`, `projectB\scoop\apps`, etc..

**Reply by rashil2000** [reply]
Are you talking about multiple parallel installations of Scoop? Because that's not really possible with the current design.

**Reply by hagaigold** [reply]
this answers my question, thx.

**Comment by hagaigold** [comment]
> Because that's not really possible with the current design

can we revisit this?

When next version #5424 will land, it is look doable with the portable config file ([feat(config): Support portable config file #5369](https://github.com/ScoopInstaller/Scoop/pull/5369)).  

I did some changes to the install script, and from initial tests it is "kind" of working- see here [`install.ps1`](https://github.com/hagaigold/scoop-Install/commit/a8eaf22c4b3c8364f17f233e9e9e17bba07142d5).  
With this install script, I can setup more than one "scoop directories", using a new switch `-Portable`.  

Of-course there are changes required in the `scoop` side:  
- don't add shimsDir to path
- don't create startup shortcuts
- more ?  

But let see first, if this is something that we want.  
What am I missing here?

**Comment by hagaigold** [comment]
If anyone want to test/play with more than one scoop location on the same machine, I have it working.  
For me it is just for: prod, dev & test environments, but I can imagine many other scenarios.  

The work is based on #5369 & #1813 (no PR yet although it is working here).  

To install a second location of `scoop`, you will need to use `install.ps1` from [here](https://github.com/hagaigold/scoop-Install).  
Clone/download it, and execute with all the three flags like in this sample:
```
.\install.ps1 -Portable -Develop -ScoopDir C:\CodeW\test-scoop
```

You will need to use repo: https://github.com/hagaigold/Scoop, with branch: **`feature/portable`**.
Navigate to your **new scoop location** (from `-ScoopDir` in previous command) into the **`shims`** directory:
```
.\scoop config scoop_branch "feature/portable"
.\scoop config
```
> Note: The `.\` before the scoop command is essential!  

You should see something like that:
```
root_path    : C:\CodeW\test-scoop
scoop_repo   : https://github.com/hagaigold/Scoop.git
scoop_branch : feature/portable
portable     : True
last_update  : 2023-10-14T20:31:10.1599406+03:00
```

Next:
```
.\scoop update
.\scoop reset -h
```

If all gone well, you should have now, some new options for the `reset` command, try it:
```
.\scoop reset --connect
```

Your new scoop location should be setup in PATH and ready to use.  
**Restart the console** - not everything is (yet) working as I wanted too.
Test it, no `.\` this time :)
```
scoop config
scoop list
scoop install ripgrep
scoop list
```
